R/as_tibble.R

```diff
@@ -110,46 +112,40 @@ as_tibble.data.frame <- function(x, validate = TRUE, ...,
 #' @export
 #' @rdname as_tibble
 as_tibble.list <- function(x, validate = TRUE, ..., .rows = NULL,
-                           .tidy_names = if (validate) NULL else FALSE) {
+                           .name_repair = NULL) {
   if (!missing(validate)) {
-    warn("The `validate` argument to `as_tibble()` is deprecated. Please use `.tidy_names` to control column names.")
+    warn("The `validate` argument to `as_tibble()` is deprecated. Please use `.name_repair` to control column names.")
+    .name_repair <- if (isTRUE(validate)) "none_passive" else "none"
```

Comment:
This might need refinement.

Reply:
Agreed - a soft deprecation might be a better strategy here.

Reply:
Same in `as_tibble.data.frame()` .

Reply:
I was talking about `.name_repair <- ...` but I suspect @hadley may be talking about the use of `validate`. Is that right? If so, what exact form of "soft depracation" do we want?

Reply:
A soft-dep here would just be to just turn the warning into a message so it doesn't break downstream packages

Reply:
How would the warning break downstream dependencies?

I'd rather be silent here for now, and start giving messages/warnings only after we have a version on CRAN where `.name_repair` works. This buys more time to adapt.